datapackage/resource_new.py

```diff
@@ -0,0 +1,267 @@
+            - multipart-local/multipart-remote -> path (str[])
+
+        Example:
+
+        ```
+        if resource.source_type == 'local':
+            open(resource.source, mode='rb').read()
+        elif resource.source_type == 'remote':
+            requests.get(resource.source).text
+        elif resource.source_type.startswith('multipart'):
+            # logic to handle list of chunks
+        ```
+
+        """
+        return self.__source
+
+    @property
+    def table(self):
+        """None/tableschema.Table: provide Table API for tabular resource
+
+        Example:
+        ```
+        if resource.table:
+            resource.schema
+            resource.iter()
+            resource.read(keyed=True)
+            resource.save('table.csv')
+        ```
+
+        Reference:
+            https://github.com/frictionlessdata/tableschema-py#table
+
+        """
+
+        # Resource -> Regular
+        if self.descriptor['profile'] != 'tabular-data-resource':
+            return None
+
+        # Resource -> Tabular
+        source = self.source
+        if self.source_type.startswith('multipart'):
+            source = _MultipartSource(self.source, self.source_type)
+        schema = self.descriptor['schema']
+        options = _get_table_options(self.descriptor)
+        table = Table(source, schema, **options)
+        return table
+
+
+# Internal
+
+_DIALECT_KEYS = [
+    'delimiter',
+    'doubleQuote',
+    'lineTerminator',
+    'quoteChar',
+    'escapeChar',
+    'skipInitialSpace',
+]
+
+
+def _get_source_with_type(data, path, base_path):
+
+    # Inline
+    if data is not None:
+        source = data
+        source_type = 'inline'
+
+    # Local/Remote
+    elif len(path) == 1:
+        if path[0].startswith('http'):
+            source = path[0]
+            source_type = 'remote'
+        elif base_path and base_path.startswith('http'):
+            source = urljoin(base_path, path[0])
+            source_type = 'remote'
+        else:
+            if not helpers.is_safe_path(path[0]):
+                raise exceptions.DataPackageException(
+                    'Local path "%s" is not safe' % path[0])
+            if not base_path:
+                raise exceptions.DataPackageException(
+                    'Local path "%s" requires base path' % path[0])
+            source = os.path.join(base_path, path[0])
+            source_type = 'local'
+
+    # Multipart Local/Remote
+    elif len(path) > 1:
+        source = []
+        source_type = 'multipart-local'
+        for chunk_path in path:
+            chunk_source, chunk_source_type = _get_source_with_type(
+                None, [chunk_path], base_path)
+            source.append(chunk_source)
+            if chunk_source_type == 'remote':
+                source_type = 'multipart-remote'
+
+    return source, source_type
+
+
+def _get_table_options(descriptor):
+
+    # General
+    options = {}
+    if not descriptor.get('data'):
```

Comment:
@amercader 
I've changed it to make more self-explaining:
```python
    options['format'] = 'csv'
    if descriptor.get('data'):
        options['format'] = 'native'
```
So per specs-v1 `resource.data` is a already native (not in a string form) array like:
```
data: [['id', 'name'], [1, 'english']]
```
`tabulator` could use it as a `source` with `format=native`. First version of code was just relaying on autodetection.

Here is an example how tabulator's source should look for `csv` format if it's not a file:
```python
tabulator.Stream('id,name\n1,english', format='csv)
```
So for our `resource.data`:
```
tabulator.Stream([['id', 'name'], [1, 'english']], format='native)
```